### ISSUE TITLE
Bugs with follow_symbolic_link set to yes

### ISSUE THREAD

**Issue by Molter73** [wazuh/wazuh#6690]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0.2 | FIM | Manager/Agent | Packages/Sources | Linux |

We've noticed the following incorrect behaviours with the refactor done on the follow_symbolic_links option in [this issue](https://github.com/wazuh/wazuh/issues/6320)

- When a symbolic link and it's corresponding hardlink are configured to be monitored, if the symbolic link is changed to a new location it removes all entries from the DB corresponding to the hardlink. When a new scan is triggered, the directory is scanned due to it being in the configuration and, since it has no entries in the DB, sends an `added` alert for each of it's subdirectories. We need to rethink this part of the code, so that if an existing configuration will take over monitoring a directory from a symlink it silently scans it instead of removing all entries.
https://github.com/wazuh/wazuh/blob/f9d06c8194ac842d201b222685d4f5a85e1e9250/src/syscheckd/run_check.c#L582-L585

- The previous case also replaces the contents of `syscheck.symbolic_links` with `'\0'` instead of a `NULL` pointer, this causes the entry to be permanently considered a symbolic link to `""`. This causes the directory to not be monitored, since an empty string if not a valid configuration. This can be fixed by properly setting `syscheck.symbolic_links` to `NULL` here:
https://github.com/wazuh/wazuh/blob/f9d06c8194ac842d201b222685d4f5a85e1e9250/src/syscheckd/run_check.c#L607

- If the opposite action is taken and a directory is replaced with a symbolic link, regardless of the value set in follow_symbolic_link, the new link will be treated as if it was configured with `follow_symbolic_link="no"`. The checker thread needs to account for this by removing the check on `syscheck.symbolic_links` in this line:
https://github.com/wazuh/wazuh/blob/f9d06c8194ac842d201b222685d4f5a85e1e9250/src/syscheckd/run_check.c#L533

Before fixing the previously mentioned bugs, we will need to create integration tests to expose each of this behaviours, in order to easily ensure they are fixed later. This integration tests must be developed as part of [this issue](https://github.com/wazuh/wazuh-qa/issues/941).

**Comment by jesusjimsa** [comment]
About the third bug, the check on `syscheck.symbolic_links` cannot just be removed as you can configure several directories in the same line and a folder (not a symbolic link) can have `follow_symbolic_link` enabled. It will enter the `else` path in the `symlink_checker_thread` function, running the `fim_link_reload_broken_link` which will discard it, but we will lose some time inside this function and cause a message to be written in the log for every folder in the `directories` tag with `follow_symbolic_link` enabled.

https://github.com/wazuh/wazuh/blob/c2ccb762b8c4c25071bd6d6980c59d4249733a24/src%2Fsyscheckd%2Frun_check.c#L539-L566

**Comment by jesusjimsa** [comment]
I have found the [`lstat` function](https://man7.org/linux/man-pages/man2/lstat.2.html) can return if a path is a symbolic link. Adding `lstat` and using the `S_IFLNK` constant to check if it is a link would allow us to remove the check on `syscheck.symbolic_links[i]` and not have to run `fim_link_reload_broken_link` for every folder configured with `follow_symbolic_link` enabled.

**Comment by jesusjimsa** [comment]
The integration tests have been failing a lot in macOS and Solaris. We have found that this is because of a bug in the double scan for `file_limit` that only affects the _scheduled_ scan. The integration tests in this Operating Systems only run in _scheduled_ and that's why they appear to only fail there.
This bug was fixed here 68187e987b2d3736afcda32985b442c665be9792, in the Windows registry pull request. To fix all the failed tests I will have to add this change to my code and we will have to fix the conflicts later.

**Comment by Molter73** [comment]
I just found another bug that causes false positve alerts. If you monitor a link with `follow_symbolic_links` set to `yes` and change its target, the query used to delete entries from the DB will not account for subdirectories that might be monitored with a different configuration. As an example, having the following configuration:
``` xml
    <directories realtime="yes" follow_symbolic_link="yes">/home/vagrant/link1</directories>
    <directories realtime="yes" follow_symbolic_link="yes">/home/vagrant/link2</directories>
```

And setting both links to point to a directory and a corresponding subdirectory, like this:
```
lrwxrwxrwx  1 vagrant vagrant      10 Dec 17 11:49 link1 -> monitored/
lrwxrwxrwx  1 vagrant vagrant      16 Dec 16 17:26 link2 -> monitored/test1/
```

If you change the target of  `link1`, all entries corresponding to `monitored/` will be deleted. When the next scan occurs, `added` alerts for the contents of `monitored/test1/` will be issued.

I will try to fix this by checking the directories to be deleted against the configuration, if I see it takes too long I will create a sepparate issue to handle this.
